api/src/public/login/script.js

```diff
@@ -153,17 +179,20 @@ document.addEventListener('DOMContentLoaded', function() {
 
   translate();
 
-  document.getElementById('login').addEventListener('click', submit, false);
+  document.getElementById('locale').addEventListener('click', handleLocaleSelection, false);
 
-  const user = document.getElementById('user');
-  user.addEventListener('keydown', focusOnPassword, false);
-  user.focus();
+  if (document.getElementById('tokenLogin')) {
+    requestTokenLogin();
+  } else {
+    document.getElementById('login').addEventListener('click', submit, false);
 
-  document.getElementById('password').addEventListener('keydown', focusOnSubmit, false);
-  
-  document.getElementById('locale').addEventListener('click', handleLocaleSelection, false);
+    const user = document.getElementById('user');
+    user.addEventListener('keydown', focusOnPassword, false);
+    user.focus();
 
-  if ('serviceWorker' in navigator) {
-    navigator.serviceWorker.register('/service-worker.js');
+    document.getElementById('password').addEventListener('keydown', focusOnSubmit, false);
+    if ('serviceWorker' in navigator) {
```

Comment:
only registering the service worker for the standard login because I saw very weird behavior when the page was loaded for a very short time and the app just hung after downloading resources (no errors).